### PR TITLE
chore(idpe-17434): remove utf8-percent encoding from onWrite path

### PR DESCRIPTION
**Pull request by wiedld**
Part of https://github.com/influxdata/idpe/issues/17434

---

* chore(idpe-17434): remove utf8-percent encoding on v2 write path, such that it matches v1 writes and onCreate ([b870242](https://github.com/influxdata/influxdb_iox/pull/7621/commits/b870242ec7f51360d22be4336eb061689a853c7c))
* chore(idpe-17434): make restrictive whitelist of chars accepted, for any NamespaceName ([daabe96](https://github.com/influxdata/influxdb_iox/pull/7621/commits/daabe9663cb197ea5e6a788536b6492f949ee6c7))

## Checklist
- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: service_grpc_namespace/src/lib.rs [hunk 1]
```diff
@@ -633,23 +633,26 @@ mod tests {
...
     );
 
     test_create_namespace_name!(
         null,
-        name = "bad \0 bananas",
+        name = "bad\0bananas",
         want = Err(e) => {
             assert_eq!(e.code(), Code::InvalidArgument);
-            assert_eq!(e.message(), "namespace name 'bad \0 bananas' contains invalid character, character number 4 is a control which is not allowed");
+            assert_eq!(e.message(), "namespace name 'bad\0bananas' contains invalid character, character number 3 is not whitelisted");
```
Comment (wiedld) [inline service_grpc_namespace/src/lib.rs:655]: It now rejects spaces.
Updated test so it was checking the null char.

**Comment by wiedld** [comment]
Note: once this merges, we have to communicate to the cst-demo clusters. Since they are using db names with whitespace and `%`.

Screen Shot 2023-04-24 at 1 28 13 PM

File: data_types/src/namespace_name.rs [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
 
 /// Length constraints for a [`NamespaceName`] name.
 ///
 /// A `RangeInclusive` is a closed interval, covering [1, 64]
 const LENGTH_CONSTRAINT: RangeInclusive<usize> = 1..=64;
 
+/// Whitelisted chars for a [`NamespaceName`] name.
+///
+/// '/' | '_' | '-' are utilized by the platforms.
+fn is_whitelisted(c: char) -> bool {
```
Comment (philjb) [inline data_types/src/namespace_name.rs:13]: nit: allowlist or passlist is preferred

File: router/src/server/http/write/single_tenant/mod.rs [hunk 3]
```diff
@@ -483,17 +483,24 @@ mod tests {
...
     test_parse_v2!(
-        no_encoding,
+        url_encoding,
         // URL-encoded input that is decoded in the HTTP layer
-        query_string = "?bucket=cool%2Fconfusing%F0%9F%8D%8C&prg=org",
+        query_string = "?bucket=cool%2Fconfusing&prg=org",
         want = Ok(WriteParams {namespace, ..}) => {
             // Yielding a not-encoded string as the namespace.
-            assert_eq!(namespace.as_str(), "cool/confusing🍌");
+            assert_eq!(namespace.as_str(), "cool/confusing");
```
Comment (philjb) [inline router/src/server/http/write/single_tenant/mod.rs:492]: no bananas

**Review by philjb — APPROVED** [review]

File: data_types/src/namespace_name.rs (hunk 2 above)
Comment (wiedld) [reply data_types/src/namespace_name.rs]: done.

**Review by philjb — APPROVED** [review]

**Review by domodwyer — APPROVED** [review]
🚀

**Pull request merged**